trace/span.proto

```diff
@@ -0,0 +1,323 @@
+      uint64 compressed_message_size = 5;
+    }
+
+    // A `TimeEvent` can contain either an `Annotation` object or a
+    // `NetworkEvent` object, but not both.
+    oneof value {
+      // One or more key:value pairs.
+      Annotation annotation = 2;
+
+      // An event describing an RPC message sent/received on the network.
+      NetworkEvent network_event = 3;
+    }
+  }
+
+  // A collection of `TimeEvent`s. A `TimeEvent` is a time-stamped annotation
+  // on the span, consisting of either user-supplied key:value pairs, or
+  // details of an RPC message sent/received on the network.
+  message TimeEvents {
+    // A collection of `TimeEvent`s.
+    repeated TimeEvent time_event = 1;
+
+    // The number of dropped annotations in all the included time events.
+    // If the value is 0, then no annotations were dropped.
+    int32 dropped_annotations_count = 2;
+
+    // The number of dropped network events in all the included time events.
+    // If the value is 0, then no network events were dropped.
+    int32 dropped_network_events_count = 3;
+  }
+
+  // The included time events.
+  TimeEvents time_events = 9;
+
+  // A pointer from the current span to another span in the same trace or in a
+  // different trace. For example, this can be used in batching operations,
+  // where a single batch handler processes multiple requests from different
+  // traces or when the handler receives a request from a different project.
+  message Link {
+    // `TRACE_ID` a unique identifier for a trace.
+    string trace_id = 1;
+
+    // `SPAN_ID` a unique identifier for a span within a trace,
+    string span_id = 2;
+
+    // The relationship of the current span relative to the linked span: child,
+    // parent, or unspecified.
+    enum Type {
+      // The relationship of the two spans is unknown.
+      TYPE_UNSPECIFIED = 0;
+      // The linked span is a child of the current span.
+      CHILD_LINKED_SPAN = 1;
+      // The linked span is a parent of the current span.
+      PARENT_LINKED_SPAN = 2;
+    }
+
+    // The relationship of the current span relative to the linked span.
+    Type type = 3;
+
+    // A set of attributes on the link.
+    Attributes attributes = 4;
+  }
+
+  // A collection of links, which are references from this span to a span
+  // in the same or different trace.
+  message Links {
+    // A collection of links.
+    repeated Link link = 1;
+    // The number of dropped links after the maximum size was enforced. If
+    // this value is 0, then no links were dropped.
+    int32 dropped_links_count = 2;
+  }
+
+  // The inclued links.
+  Links links = 10;
+
+  // An optional final status for this span.
+  io.opencensus.trace.proto.Status status = 11;
+
+  // A highly recommended but not required flag that identifies when a trace
+  // crosses a process boundary. True when the parent_span belongs to the
+  // same process as the current span.
+  google.protobuf.BoolValue same_process_as_parent_span = 12;
```

Comment:
is this expected to be used (ex shared span zipkin v1 style?)

Reply:
this is to replace the notion of span type (client/server/local). We think that in case of proxies for example you don't want to necessary create 2 spans to fit that model.